### DISCUSSION TITLE
Manipulating supported OS versions in csproj file

### DISCUSSION THREAD

**Discussion by HobDev** [dotnet/maui#3615] [Q&A]
These are the tags used for defining OS version supported by maui app:

```
<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-ios'">15.0</SupportedOSPlatformVersion>
<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-maccatalyst'">15.0</SupportedOSPlatformVersion>
<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-android'">31.0</SupportedOSPlatformVersion>
<SupportedOSPlatformVersion Condition="$(TargetFramework.Contains('-windows'))">10.0.19041.0</SupportedOSPlatformVersion>
<TargetPlatformMinVersion Condition="$(TargetFramework.Contains('-windows'))">10.0.17763.0</TargetPlatformMinVersion>

```

But how to define min version and target android version on Android in csproj file.  The min version is only defined for windows over here. But not for iOS, Android and Mac

**Comment by mas985** [comment]
That is specified in the Platforms/Android/AndroidManifest.xml file

	<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="31" />

**Comment by HobDev** [comment]
@mas985 
Ok, but what is the purpose of  `SupportedOSPlatformVersion`  in the csproj. It should reflect the configuration of the platforms, but changing the vesions in Android project does not change the csproj.

**Comment by mas985** [comment]
From my somewhat limited understanding (not great documentation), SupportedOSPlatformVersion is used by the IDE to determine the .NET library for compilation. I have tried increasing this value greater than the phone OS but the app still runs fine on the phone so I am not exactly sure what it does but I do know that the library does not support devices below Android 21 which is the default for the MAUI project. 

Where as minSdkVersion is used by the Android build tools and Play Store to define the minimum OS version for installation. Increasing the minSdkVersion has the effect of reducing the number of devices that Play Store allows the app to be installed which I have verified works to reduce the number of devices compatible with the app. The manifest dictates which apps can be installed on which devices so only those devices can see the apps in Play Store.

So basically one is for the IDE and the other for the Android ecosystem.

A few references:

https://docs.microsoft.com/en-us/dotnet/maui/supported-platforms
https://docs.microsoft.com/en-us/visualstudio/ide/visual-studio-multi-targeting-overview?view=vs-2022

What exactly are you trying to accomplish? Why not use the defaults minimums for both as 21?

One more reference:

https://docs.microsoft.com/en-us/dotnet/standard/frameworks#support-older-os-versions

> Although a platform-specific app or library is compiled against APIs from a specific version of that OS, you can make it compatible with earlier OS versions by adding the SupportedOSPlatformVersion property to your project file. The SupportedOSPlatformVersion property indicates the minimum OS version required to run your app or library. If you don't explicitly specify this minimum run-time OS version in the project, it defaults to the platform version from the TFM.
